app/connectors/VatApiConnector.scala

```diff
@@ -34,7 +35,7 @@ class VatApiConnector @Inject()(http: HttpClient, appConfig: AppConfig) {
 
   private[connectors] def obligationsUrl(vrn: String): String = s"${appConfig.vatApiBaseUrl}/$vrn/obligations"
   private[connectors] def customerInfoUrl(vrn: String): String = s"${appConfig.vatApiBaseUrl}/customer-information/vat/$vrn"
-  private[connectors] def returnUrl(vrn: String, periodKey: String) = s"${appConfig.vatApiBaseUrl}/$vrn/returns/$periodKey"
+  private[connectors] def returnUrl(vrn: String, periodKey: String) = s"${appConfig.vatApiBaseUrl}/$vrn/returns/${URLEncoder.encode(periodKey)}"
```

Comment:
Can you change the `URLEncoder.encode(periodKey)` to `URLEncoder.encode(periodKey, "UTF-8")`. The former is now deprecated and generating a warning during the compile.